### PR TITLE
New http-request agent

### PR DESCRIPTION
**Pull request by nicoloboschi**
* Added new `http-request` agent
* It accepts
  * url: http(s) url to call without query-string
  * query-string (map): accepts string as template. encoding is done automatically
  * method: "get" by default
  * headers (map):  accepts string as template e.g. X-Custom-Header: {{{ value.id }}}
  * body: body to send, must be a string:  accepts string as template e.g. X-Custom-Header: {{{ value.id }}}
  * output-field: field where to output in the record. in case the response return content-type: application/json it will be deserialized as map. otherwise will be threated as string 
  * allow-redirects: set to false to never redirect
  * handle-cookies: set to false to not manage cookies during redirects

Example with authentication
```
                                  - name: "http-request"
                                    type: "http-request"
                                    input: input-topic
                                    output: output-topic
                                    id: step1
                                    configuration:
                                        output-field: value.api
                                        url: "https://somewhere/api/test/{{{ value.id }}}"
                                        method: POST
                                        body: '{"id": "{{{ value.id }}}"}'
                                        headers:
                                          Content-Type: application/json
                                          Authorization: Bearer {{{ secrets.s1.token }}}
```

### CODE REVIEW

**Review by eolivelli — CHANGES_REQUESTED** [review]
I would make the query string parameters configurable, otherwise you have to provide a way to perform escaping.

Also I would make the url a constant in the first iteration, this way we prevent malicious usage, like calling URLs forged by a chatbot user

File: langstream-agents/langstream-agent-http-request/src/main/java/ai/langstream/agents/queryhttp/HttpRequestAgent.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    @Override
+    public void init(Map<String, Object> configuration) {
+        final String url =
+                ConfigurationUtils.requiredNonEmptyField(
+                        configuration, "url", () -> "http-request agent");
+        this.outputFieldName =
+                ConfigurationUtils.requiredNonEmptyField(
+                        configuration, "output-field", () -> "http-request agent");
+        this.method = ConfigurationUtils.getString("method", "GET", configuration);
+        final String body = ConfigurationUtils.getString("body", null, configuration);
```
Comment (eolivelli) [inline langstream-agents/langstream-agent-http-request/src/main/java/ai/langstream/agents/queryhttp/HttpRequestAgent.java:72]: pleae make the user agent configurable, also the handling of redirections and cookies, like we do in the webcrawler

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by cbornet** [comment]
It can be interesting to look at what features LangStream loaders provide when we create these Sources.

**Pull request merged**